modules/text2vec-openai/vectorizer/objects.go

```diff
@@ -99,3 +118,113 @@ func (v *Vectorizer) getVectorizationConfig(cfg moduletools.ClassConfig) ent.Vec
 		Dimensions:   settings.Dimensions(),
 	}
 }
+
+// batchWorker is a go routine that handles the communication with the vectorizer
+//
+// On the high level it has the following steps:
+//  1. It receives a batch job
+//  2. It splits the job into smaller vectorizer-batches if the token limit is reached. Note that objects from different
+//     batches are not mixed with each other.
+//  3. It sends the smaller batches to the vectorizer
+func (v *Vectorizer) batchWorker() {
+	tokenLimit := 1000 // don't know limits before the first request
+	remainingRequests := 10
+	requestLimitReset := 0
+	texts := make([]string, 0, 100)
+	for job := range v.jobQueueCh {
+		conf := v.getVectorizationConfig(job.cfg)
+		objCounter := 0
+		vecBatchOffset := 0 // index of first object in current vectorizer-batch
+
+		tokensInCurrentBatch := 0
+		texts = texts[:0]
+		for {
```

Comment:
can be changed to `for objCounter := 0; objCounter < len(job.objects); objCounter++`?

Reply:
That does not work, because an object might hit the token limit and we might need to send a batch and then add it to the next one